IoT-Sample/Swift/IoTSampleSwift/ConnectionViewController.swift

```diff
@@ -19,207 +19,273 @@ import AWSMobileClient
-                            self.iot.attachPrincipalPolicy(attachPrincipalPolicyRequest!).continueWith (block: { (task) -> AnyObject? in
-                                if let error = task.error {
-                                    print("failed: [\(error)]")
-                                }
-                                print("result: [\(String(describing: task.result))]")
-                                
-                                // Connect to the AWS IoT platform
-                                if (task.error == nil)
-                                {
-                                    DispatchQueue.main.asyncAfter(deadline: .now()+2, execute: {
-                                        self.logTextView.text = "Using certificate: \(certificateId!)"
-                                        self.iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:certificateId!, statusCallback: mqttEventCallback)
+    func handleDisconnect() {
+        let tabBarViewController = tabBarController as! IoTSampleTabBarController
+        self.connectButton.isHidden = false
+        self.connectIoTDataWebSocket.isHidden = false
+        activityIndicatorView.startAnimating()
+        logTextView.text = "Disconnecting..."
 
-                                    })
-                                }
-                                return nil
-                            })
-                        }
-                        else
-                        {
-                            DispatchQueue.main.async {
-                                sender.isEnabled = true
-                                self.activityIndicatorView.stopAnimating()
-                                self.logTextView.text = "Unable to create keys and/or certificate, check values in Constants.swift"
-                            }
-                        }
-                    } )
-                }
+        DispatchQueue.global(qos: DispatchQoS.QoSClass.default).async {
+            self.iotDataManager.disconnect();
+            DispatchQueue.main.async {
+                self.connected = false
+                tabBarViewController.viewControllers = [ self, self.configurationViewController ]
             }
-            else
-            {
-                let uuid = UUID().uuidString;
+        }
+    }
 
-                // Connect to the AWS IoT service
-                iotDataManager.connect( withClientId: uuid, cleanSession:true, certificateId:certificateId!, statusCallback: mqttEventCallback)
+    func searchForExistingCertificateIdInBundle() -> String? {
+        let defaults = UserDefaults.standard
+        // No certificate ID has been stored in the user defaults; check to see if any .p12 files
+        // exist in the bundle.
+        let myBundle = Bundle.main
+        let myImages = myBundle.paths(forResourcesOfType: "p12" as String, inDirectory:nil)
+        let uuid = UUID().uuidString
+        
+        if (myImages.count > 0) {
```

Comment:
Change to `guard let cert = myImages.first else { ... exit condition ... }`, and then you don't have to check for length, or dereference `myImages[0]` below

Reply:
I specifically didn't want to update this code, as this was copy and pasted from before.  Sorry, not going to update this.

Reply:
Updated!